libs/store/src/lib/infrastructure/persistence/query/query.ts

```diff
@@ -148,7 +153,10 @@ export class Query<TModel> {
 
         const params = this._createQueryParams();
         const query = this.adapter.createQueryString(params);
-        return this.service.getWithQuery(query);
+        this.service.getWithQuery(query).pipe(take(1)).subscribe(entities => {
+            this.result$.next(entities);
+        });
```

Comment:
We need to keep in mind that the previous `getWithQuery` request might be still in progress so we need to cancel the previous one and run the new one (to avoid the race conditions).